packages/runner/injection/patches/cookies.ts

```diff
@@ -0,0 +1,133 @@
+import {
+  CookieJar,
+  toughCookieToAutomationCookie,
+  automationCookieToToughCookie,
+} from '@packages/server/lib/util/cookies'
+import { Cookie as ToughCookie } from 'tough-cookie'
+import type { AutomationCookie } from '@packages/server/lib/automation/cookies'
+
+const parseDocumentCookieString = (documentCookieString: string): AutomationCookie[] => {
+  if (!documentCookieString || !documentCookieString.trim().length) return []
+
+  return documentCookieString.split(';').map((cookieString) => {
+    const [name, value] = cookieString.split('=')
+
+    return {
+      name: name.trim(),
+      value: value.trim(),
+      domain: location.hostname,
+      expiry: null,
+      httpOnly: false,
+      maxAge: null,
+      path: null,
+      sameSite: 'lax',
+      secure: false,
+    }
+  })
+}
+
+const sendCookieToServer = (cookie: AutomationCookie) => {
+  window.top!.postMessage({
+    event: 'cross:origin:aut:set:cookie',
+    data: {
+      cookie,
+      url: location.href,
+      // url will always match the cookie domain, so strict context tells
+      // tough-cookie to allow it to be set
+      sameSiteContext: 'strict',
+    },
+  }, '*')
+}
+
+// document.cookie monkey-patching
+// -------------------------------
+// We monkey-patch document.cookie when in a cross-origin injection, because
+// document.cookie runs into cross-origin restrictions when the AUT is on
+// a different origin than top. The goal is to make it act like it would
+// if the user's app was run in top.
+//
+// The general strategy is:
+// - Keep the document.cookie value (`documentCookieValue`) available so
+//   the document.cookie getter can synchronously return it.
+// - Optimistically update that value when document.cookie is set, so that
+//   subsequent synchronous calls to get the value will work.
+// - On an interval, get the browser's cookies for the given domain, so that
+//   updates to the cookie jar (via http requests, cy.setCookie, etc) are
+//   reflected in the document.cookie value.
```

Comment:
Does this comment need to be updated? I don't think we still get cookies on an interval